wm_dll.c

```diff
@@ -13,20 +23,150 @@ __declspec(dllexport) LRESULT CALLBACK ShellProc(int code, WPARAM wparam, LPARAM
 	return CallNextHookEx(NULL, code, wparam, lparam);
 }
 
+LRESULT CALLBACK WndProc(HWND hwnd, UINT msg, WPARAM wParam, LPARAM lParam) {
+    switch (msg) {
+        case WM_HOTKEY:
+			DEBUG_PRINT("DEBUG wm_dll.c L26: wParam = %i\n", wParam);  
+            switch (wParam) {
+                case WORKSPACE_1:
+					puts("Switch to workspace 1"); //TODO 
+                    break;
+                case WORKSPACE_2:
+					puts("Switch to workspace 2"); //TODO 
+                    break;
+				case WORKSPACE_3:
+					puts("Switch to workspace 3"); //TODO
+					break; 
+				case WORKSPACE_4:
+					puts("Switch to workspace 4"); //TODO 
+					break; 
+				case WINDOW_UP:
+					puts("Focus on window above current"); //TODO 
+					break; 
+				case WINDOW_DOWN:
+					puts("Focus on window below current"); //TODO
+					break;
+				case WINDOW_LEFT:
+					puts("Focus on window to the left of current"); //TODO 
+					break;
+				case WINDOW_RIGHT:
+					puts("Focus on window to the right of current"); //TODO
+					break;
+            }
+            break;
+        default:
+            return DefWindowProc(hwnd, msg, wParam, lParam);
+    }
+	
+    return 0;
+}
 
-BOOL APIENTRY DllMain(HMODULE hModule, DWORD ulReasonForCall, LPVOID lpReserved) 
+/**
+ * This is terrible... but I cannot think of any other way to do it with the RegisterHotkey function
```

Comment:
Instead of opening a new window, how about installing another hook for key presses similar to how the shell hook is installed?